iris/src/iris/retrieval/lecture/lecture_retrieval.py

```
@@ -104,18 +106,18 @@ def __init__(self, client: WeaviateClient):
         course_id: int,
         chat_history: List[PyrisMessage],
-        problem_statement: str = None,
-        exercise_title: str = None,
-        lecture_id: int = None,
-        lecture_unit_id: int = None,
-        base_url: str = None,
+        problem_statement: str | None = None,
+        exercise_title: str | None = None,
+        lecture_id: int | None = None,
+        lecture_unit_id: int | None = None,
+        base_url: str | None = None,
```

Ruff (0.14.5)

Unused method argument: `base_url`

(ARG002)